sandwich/src/Test/Sandwich/Interpreters/PruneTree.hs

```diff
@@ -0,0 +1,49 @@
+
+-- | Prunes any nodes and their children in a spec tree that match any of a list of names
+
+module Test.Sandwich.Interpreters.PruneTree (pruneTree) where
+
+import Control.Monad.Free
+import qualified Data.List as L
+import Test.Sandwich.Types.Spec
+
+pruneTree :: Free (SpecCommand context m) () -> [String] -> Free (SpecCommand context m) ()
```

Comment:
It's worth noting that `filterTree` takes `String` instead of `[String]`, so the caller is expected to `foldl` this to handle multiple filters. Might be good to do this the same way for consistency. I kind of like that the fold makes it clear at the callsite that pruning with multiple arguments is applied sequentially. The current `pruneTree` type signature is consistent with wrong interpretations, like that it only prunes nodes that contain *all* of the arguments. Not a huge deal, but I like making the picture as clear as possible.

Reply:
The reason I did it that way was that it seemed more efficient, since the list of prune labels would likely be smaller than the tree, so walking the tree once rather than for each label seemed more efficient. 

However, the performance impact is likely irrelevantly small here, so I've refactored it to your more semantically clear version.